api/models/series.go

```diff
@@ -26,19 +26,46 @@ type Series struct {
 }
 
 func (s *Series) SetTags() {
-	tagSplits := strings.Split(s.Target, ";")
+	numTags := strings.Count(s.Target, ";")
+
+	if s.Tags == nil {
+		// +1 for the name tag
+		s.Tags = make(map[string]string, numTags)
+	} else {
+		for k := range s.Tags {
+			delete(s.Tags, k)
+		}
+	}
+
+	if numTags == 0 {
+		s.Tags["name"] = s.Target
+		return
+	}
+
+	index := strings.IndexByte(s.Target, ';')
+	name := s.Target[0:index]
 
-	s.Tags = make(map[string]string, len(tagSplits))
+	remainder := s.Target
+	for index > 0 {
+		remainder = remainder[index+1:]
+		index = strings.IndexByte(remainder, ';')
 
-	for _, tagPair := range tagSplits[1:] {
-		parts := strings.SplitN(tagPair, "=", 2)
-		if len(parts) != 2 {
+		tagPair := remainder
+		if index > 0 {
+			tagPair = remainder[:index]
+		}
+
+		equalsPos := strings.IndexByte(tagPair, '=')
+		if equalsPos < 1 {
 			// Shouldn't happen
 			continue
 		}
-		s.Tags[parts[0]] = parts[1]
+
+		s.Tags[tagPair[0:equalsPos]] = tagPair[equalsPos+1:]
 	}
-	s.Tags["name"] = tagSplits[0]
+
+	// Do this last to overwrite any invalid "name" tag that might be preset
+	s.Tags["name"] = name
```

Comment:
what do you mean with invalid? one specified by the user? maybe we should refer to those as 'illegal' or even clarify as `overwrite any "name" tag that may have been illegally specified in the series tags`. or actually drop the 'illegal' because i don't think it's written anywhere that it's illegal.